### PR TITLE
Added /job/:jobId/cancel route

### PR DESCRIPTION
**Pull request by JTSIV1**
- Cancels jobs by their jobID
- If job is not running it waits for the job to enter the HPC to cancel it

### CODE REVIEW

File: server.ts [hunk 1]
```diff
@@ -1052,6 +1087,14 @@ app.post("/job/:jobId/submit", async function (req, res) {
   res.json(Helper.job2object(job));
 });
 
+/**
+ * @openapi
+ * /clean:
+ *  put:
+ *      description: Not yet implemented
+*/
+app.put("/clean", async function (req, res) {});
```
Comment (alexandermichels) [inline server.ts:1096]: It seems like we can remove this from the PR since it doesn't actually implement anything anyways.

File: server.ts [hunk 2]
```diff
@@ -516,6 +526,31 @@ app.get("/allowlist", function (req, res) {
...
+/**
+ * @openapi
+ * /announcement:
+ *  get:
+ *      description: Returns list of current announcements (Authentication NOT REQUIRED)
+ *      responses:
+ *          200:
+ *              description: Returns array of current announcements
+ */
+app.get("/announcement", function (req, res) {
```
Comment (alexandermichels) [inline server.ts:538]: In the future please do these PRs as separate branches. It seems this PR contains all of the commits from the announcement route PR.

**Review by alexandermichels — APPROVED** [review]
The code mostly looks fine, but we will test how it runs Wednesday.

File: server.ts [hunk 3]
```diff
@@ -1072,9 +1115,51 @@ app.put("/job/:jobId/resume", async function (req, res) {});
...
+ *          200:
+ *              description: Job was found successfully added to the queue to be canceled
+ *          401:
+ *              description: Returns "submit without login is not allowed" if the user is not logged in or "invalid access token" if the events cannot be accessed
+ *          402:
+ *              description: Returns 'invalid input' and a list of errors with the format of the req body - jobId may be invalid or job may not be in queue
 */
-app.put("/job/:jobId/cancel", async function (req, res) {});
+app.put("/job/:jobId/cancel", async function (req, res) {
+  console.log("made it to cancel");
```
Comment (mitkotak) [inline server.ts:1128]: Is it possible to emit messages in addition to console.logs for better debugging on the sdk side ?

**Review by mitkotak — APPROVED** [review]
Good job ! The emit stuff is optional

**Pull request merged**